Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/qrcode/fullscreen/QrCodeFullScreenFragment.kt

```diff
@@ -0,0 +1,124 @@
+package de.rki.coronawarnapp.ui.qrcode.fullscreen
+
+import android.graphics.Color
+import android.os.Build
+import android.os.Bundle
+import android.view.View
+import android.view.View.SYSTEM_UI_FLAG_FULLSCREEN
+import android.view.View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+import android.view.View.SYSTEM_UI_FLAG_IMMERSIVE
+import android.view.View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+import android.view.View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+import android.view.View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+import android.view.View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
+import android.view.View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+import android.view.View.SYSTEM_UI_FLAG_VISIBLE
+import android.view.animation.AccelerateInterpolator
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
+import com.google.android.material.transition.MaterialContainerTransform
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.FragmentQrCodeFullScreenBinding
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import javax.inject.Inject
+
+class QrCodeFullScreenFragment : Fragment(R.layout.fragment_qr_code_full_screen), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+
+    private val binding: FragmentQrCodeFullScreenBinding by viewBindingLazy()
+    private val args by navArgs<QrCodeFullScreenFragmentArgs>()
+    private val viewModel: QrCodeFullScreenViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as QrCodeFullScreenViewModel.Factory
+            factory.create(
+                qrcodeText = args.qrCodeText
+            )
+        }
+    )
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val containerTransform = MaterialContainerTransform()
+            .apply {
+                scrimColor = Color.TRANSPARENT
+                interpolator = AccelerateInterpolator()
+            }
+        sharedElementEnterTransition = containerTransform
+        sharedElementReturnTransition = containerTransform
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) =
+        with(binding) {
+            toolbar.setNavigationOnClickListener { popBackStack() }
+            qrCodeImage.setOnClickListener { viewModel.switchImmersiveMode() }
```

Comment:
Should we switch between non/immersive mode by tapping anywhere? I tapped a few times below the qr code with my thumb (easier to reach), to then notice i can only leave immersive mode when pressing the qr code itself. This may catch some users off guard.